scipy/optimize/_constraints.py

```diff
@@ -166,23 +166,33 @@ class Bounds:
 
     Parameters
     ----------
-    lb, ub : array_like
+    lb, ub : array_like, optional
         Lower and upper bounds on independent variables. Each array must
         have the same size as x or be a scalar, in which case a bound will be
         the same for all the variables. Set components of `lb` and `ub` equal
         to fix a variable. Use ``np.inf`` with an appropriate sign to disable
         bounds on all or some variables. Note that you can mix constraints of
         different types: interval, one-sided or equality, by setting different
-        components of `lb` and `ub` as necessary.
+        components of `lb` and `ub` as necessary. Defaults to lb = 0 and
+        ub = np.inf (non-negativity).
```

Comment:
Non-negativity constraints are pretty standard for LP.
This may not be standard for nonlinear programming, but if the variables are free, then the user doesn't need to specify bounds at all.

Reply:
For non-linear minimisation having `lb=-np.inf` makes much more sense, and I think having `lb=0` as a default would trip a lot of people up.

Reply:
I feel pretty strongly about the `lb=-np.inf` being the default option.

Reply:
`PreparedConstraint.__init__` is also assuming that `lb=-np.inf`.

Reply:
Right, that's why I wrote
> This may not be standard for nonlinear programming

We may have to make a compromise here because the conventions tend to be different between LP and NLP. 
But we might actually want the default for `milp` to be unbounded after all - I'm checking on that.

Reply:
I changed the defaults to `lb=-np.inf`, `ub=np.inf`. Perhaps the default for `milp` will still be non-negativity constraints, but then to make variables free the user can just provide a default `Bounds` object.